crates/ruff_python_formatter/src/expression/expr_bin_op.rs

```diff
@@ -263,10 +264,23 @@ impl NeedsParentheses for ExprBinOp {
     fn needs_parentheses(
         &self,
         parent: AnyNodeRef,
-        _context: &PyFormatContext,
+        context: &PyFormatContext,
     ) -> OptionalParentheses {
         if parent.is_expr_await() && !self.op.is_pow() {
             OptionalParentheses::Always
+        } else if let Expr::Constant(constant) = self.left.as_ref() {
+            // Multiline strings are guaranteed to never fit, avoid adding unnecessary parentheses
+            if !constant.value.is_implicit_concatenated()
+                && is_multiline_string(constant, context.source())
+                && has_own_parentheses(&self.right, context).is_some()
+                && !context.comments().has_dangling(self)
+                && !context.comments().has(self.left.as_ref())
+                && !context.comments().has(self.right.as_ref())
```

Comment:
I added a test for parenthesized right sides but decided to keep the formatting as is if the right side contains any leading comments because our current binary expression formatting would format the above as

```python
expected_content = 
    """<?xml version="1.0" encoding="UTF-8"?>
<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
<sitemap><loc>%s/simple/sitemap-simple.xml</loc><lastmod>%s</lastmod>
</sitemap>
</sitemapindex>
"""
%
(
      # Doesnt need parentheses
      self.base_url
)
```

Which results in a syntax error. I haven't seen this in any real project yet and don't think it justifies implementing an entire new binary expression layout.